### PR TITLE
switch from OAuth2 to GitHub App authentication

### PR DESCRIPTION
**Pull request by lmcmicu**
This changes the authentication process over from OAuth2 authentication to GitHub Apps authentication. I've verified that the basic authentication process is working and the API calls appear to be working fine too.

@jamesaoverton and @jannahastings : Since it has been a very long time since I have worked on the `purl-editor` I'm not very confident in my ability to exhaustively test the rest of actual the editor functionality to make sure that all is still working as expected. It should be fine, once you set up an appropriate GitHub App with the right permissions. But I don't know which permissions to set, or which repos to give it access to. I'm also wary of editing any files without knowing what I'm doing. Can you have a look please?

### CODE REVIEW

File: config.py
Comment (jamesaoverton) [inline config.py]: This is fine as a default, but I will need to be able to set the server base with an environment variable. I don't see where the code is reading `SERVER_BASE` from the environment.

**Comment by jamesaoverton** [comment]
Looks good! I'll try this out on the dev server as soon as I get a chance.

I don't think this will interfere with any of @jannahastings's work.

File: config.py
Comment (lmcmicu) [reply config.py]: It isn't reading from the environment, but you're right that that would be better. I'll change it.

**Comment by jannahastings** [comment]
> Looks good! I'll try this out on the dev server as soon as I get a chance.
> 
> I don't think this will interfere with any of @jannahastings's work.

Yes it looks good to me! I'll wait for the OK from @jamesaoverton  and then merge it into my working branch.

File: config.py
Comment (lmcmicu) [reply config.py]: Done.

File: server.py
Comment (jamesaoverton) [inline server.py]: I got an error, maybe from here: `github_call() got an unexpected keyword argument 'data'`

**Review by jamesaoverton — CHANGES_REQUESTED** [review]
Sorry for the delay. I think I found a small bug: should the named argument for POST be `data` or `params`?

File: server.py
Comment (lmcmicu) [reply server.py]: Yes it should be `params`. Thanks for spotting that. It's wrong in more than one place, actually. As I mentioned in the PR description I wasn't able to back-test all of the old functionality. Sorry about that. I'll push a fix in a minute.

Comment (lmcmicu) [reply server.py]: Try it again now.

**Review by jamesaoverton — APPROVED** [review]

**Comment by jamesaoverton** [comment]
Thanks @lmcmicu, that fixed the problem!

@jannahastings Please merge this when you're ready.

**Comment by lmcmicu** [comment]
Oh, I'm sorry! @jamesaoverton asked @jannahastings to merge this, but I thought he was addressing me. I hope I didn't cause any problems by doing so.

**Comment by jannahastings** [comment]
No problem! I have to merge it into my branch as well :-)

**Pull request merged**
